lib/iex/lib/iex/introspection.ex

```diff
@@ -557,26 +547,60 @@ defmodule IEx.Introspection do
       :error ->
         :no_beam
 
-      _ when is_nil(docs) ->
-        :no_docs
-
       {:ok, callbacks} ->
         docs =
-          docs
+          callbacks
+          |> Enum.map(&translate_callback/1)
           |> Enum.filter(filter)
-          |> Enum.map(fn
-            {{:macrocallback, fun, arity}, _, _, doc, metadata} ->
-              macro = {:"MACRO-#{fun}", arity + 1}
-              {format_callback(:macrocallback, fun, macro, callbacks), doc, metadata}
-
-            {{kind, fun, arity}, _, _, doc, metadata} ->
-              {format_callback(kind, fun, {fun, arity}, callbacks), doc, metadata}
+          |> Enum.sort()
+          |> Enum.reduce([], fn {{_, function, arity}, _specs} = callback, acc ->
```

Comment:
I think this could be a `flat_map`. And in that case there also wouldn't be a need to reverse at the end.

Reply:
Sounds good to me! Done